plugins/mapper-annotated-text/src/main/java/org/elasticsearch/search/fetch/subphase/highlight/AnnotatedTextHighlighter.java

```diff
@@ -37,39 +38,37 @@
 
     public static final String NAME = "annotated";
 
-    @Override
-    protected Analyzer getAnalyzer(DocumentMapper docMapper, HitContext hitContext) {
-        return new AnnotatedHighlighterAnalyzer(super.getAnalyzer(docMapper, hitContext), hitContext);
-    }
-
     // Convert the marked-up values held on-disk to plain-text versions for highlighting
     @Override
-    protected List<Object> loadFieldValues(MappedFieldType fieldType,
-                                           Field field,
-                                           HitContext hitContext,
-                                           boolean forceSource) throws IOException {
-        List<Object> fieldValues = super.loadFieldValues(fieldType, field, hitContext, forceSource);
-        String[] fieldValuesAsString = fieldValues.toArray(new String[fieldValues.size()]);
+    protected List<Object> loadFieldValues(
+        CustomUnifiedHighlighter highlighter,
+        MappedFieldType fieldType,
+        Field field,
+        HitContext hitContext,
+        boolean forceSource
+    ) throws IOException {
+        List<Object> fieldValues = super.loadFieldValues(highlighter, fieldType, field, hitContext, forceSource);
 
-        AnnotatedText[] annotations = new AnnotatedText[fieldValuesAsString.length];
-        for (int i = 0; i < fieldValuesAsString.length; i++) {
-            annotations[i] = AnnotatedText.parse(fieldValuesAsString[i]);
+        List<Object> strings = new ArrayList<>(fieldValues.size());
+        AnnotatedText[] annotations = new AnnotatedText[fieldValues.size()];
+        for (int i = 0; i < fieldValues.size(); i++) {
+            annotations[i] = AnnotatedText.parse(fieldValues.get(i).toString());
+            strings.add(annotations[i].textMinusMarkup);
         }
-        // Store the annotations in the hitContext
-        hitContext.cache().put(AnnotatedText.class.getName(), annotations);
+        // Store the annotations in the formatter and analyzer
+        ((AnnotatedPassageFormatter) highlighter.getFormatter()).setAnnotations(annotations);
+        ((AnnotatedHighlighterAnalyzer) highlighter.getIndexAnalyzer()).setAnnotations(annotations);
+        return strings;
+    }
 
-        ArrayList<Object> result = new ArrayList<>(annotations.length);
-        for (int i = 0; i < annotations.length; i++) {
-            result.add(annotations[i].textMinusMarkup);
-        }
-        return result;
+    @Override
```

Comment:
I think this change means we can remove `hitContext` from the method sig here as well - we only added it in to access the cache.